src/main.rs

```diff
@@ -647,13 +878,131 @@ fn get_setup(args: &[String]) -> Setup {
         )
     };
 
-    if credentials.is_none() && matches.opt_present(DISABLE_DISCOVERY) {
+    let enable_discovery = !matches.opt_present(DISABLE_DISCOVERY);
+
+    if credentials.is_none() && !enable_discovery {
         error!("Credentials are required if discovery is disabled.");
         exit(1);
     }
 
+    if !enable_discovery && matches.opt_present(ZEROCONF_PORT) {
+        warn!(
+            "With the `--{}` / `-{}` flag set `--{}` / `-{}` has no effect.",
+            DISABLE_DISCOVERY, DISABLE_DISCOVERY_SHORT, ZEROCONF_PORT, ZEROCONF_PORT_SHORT
+        );
+    }
+
+    let zeroconf_port = if enable_discovery {
+        matches
+            .opt_str(ZEROCONF_PORT)
+            .map(|port| {
+                let on_error = || {
+                    error!(
+                        "Invalid `--{}` / `-{}`: {}",
+                        ZEROCONF_PORT, ZEROCONF_PORT_SHORT, port
+                    );
+                    println!(
+                        "Valid `--{}` / `-{}` values: 1 - 65535",
+                        ZEROCONF_PORT, ZEROCONF_PORT_SHORT
+                    );
+                };
+
+                let port = port.parse::<u16>().unwrap_or_else(|_| {
+                    on_error();
+                    exit(1);
+                });
+
+                if port == 0 {
+                    on_error();
+                    exit(1);
+                }
+
+                port
+            })
+            .unwrap_or(0)
+    } else {
+        0
+    };
+
+    let connect_config = {
+        let connect_default_config = ConnectConfig::default();
+
+        let name = matches
+            .opt_str(NAME)
+            .unwrap_or_else(|| connect_default_config.name.clone());
+
+        let initial_volume = matches
+            .opt_str(INITIAL_VOLUME)
+            .map(|initial_volume| {
+                let on_error = || {
+                    error!(
+                        "Invalid `--{}` / `-{}`: {}",
+                        INITIAL_VOLUME, INITIAL_VOLUME_SHORT, initial_volume
+                    );
+                    println!(
+                        "Valid `--{}` / `-{}` values: 0 - 100",
+                        INITIAL_VOLUME, INITIAL_VOLUME_SHORT
+                    );
+                    #[cfg(feature = "alsa-backend")]
+                    println!(
+                        "Default: {}, or the current value when the Alsa mixer is used.",
```

Comment:
Small nit, we're using "alsa" and "Alsa" interchangeably in various lines.

Reply:
I can change everything to "alsa"

Reply:
Done.